File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        Set<Integer> set = new HashSet<>();
        int ans = 0;
        int index = 0;
        
        for(int num : nums){
            if(!set.contains(num)) {
                set.add(num);
                ans++;
                nums[index++] = num;
            }   
        }

        return ans;
    }
}

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        curind = 1
        curval = nums[0]

        n = len(nums) 

        for i in range(1,n,1):
            if curval == nums[i] :
                continue
            nums[curind] = nums[i]
            curind += 1
            curval = nums[i]
        return curind